modules/percolator/src/main/java/org/elasticsearch/percolator/ExtractQueryTermsService.java

```diff
@@ -64,35 +65,48 @@
+    public static final String EXTRACTION_PARTIAL = "partial";
+    public static final String EXTRACTION_FAILED = "failed";
 
     private ExtractQueryTermsService() {
     }
 
     /**
      * Extracts all terms from the specified query and adds it to the specified document.
-     * @param query                 The query to extract terms from
-     * @param document              The document to add the extracted terms to
-     * @param queryTermsFieldField  The field in the document holding the extracted terms
-     * @param unknownQueryField     The field used to mark a document that not all query terms could be extracted.
-     *                              For example the query contained an unsupported query (e.g. WildcardQuery).
-     * @param fieldType The field type for the query metadata field
+     * @param query                     The query to extract terms from
+     * @param document                  The document to add the extracted terms to
+     * @param queryTermsFieldField      The field in the document holding the extracted terms
+     * @param extractionResultField     The field contains whether query term extraction was successful, partial or
+     *                                  failed. (For example the query contained an unsupported query (e.g. WildcardQuery)
+     *                                  then query extraction would fail)
+     * @param fieldType                 The field type for the query metadata field
      */
     public static void extractQueryTerms(Query query, ParseContext.Document document, String queryTermsFieldField,
-                                         String unknownQueryField, FieldType fieldType) {
-        Set<Term> queryTerms;
+                                         String extractionResultField, FieldType fieldType) {
+        Result result;
         try {
-            queryTerms = extractQueryTerms(query);
+            result = extractQueryTerms(query);
         } catch (UnsupportedQueryException e) {
-            document.add(new Field(unknownQueryField, new BytesRef(), fieldType));
+            document.add(new Field(extractionResultField, EXTRACTION_FAILED, fieldType));
             return;
         }
-        for (Term term : queryTerms) {
+        for (Term term : result.terms) {
             BytesRefBuilder builder = new BytesRefBuilder();
             builder.append(new BytesRef(term.field()));
             builder.append(FIELD_VALUE_SEPARATOR);
             builder.append(term.bytes());
             document.add(new Field(queryTermsFieldField, builder.toBytesRef(), fieldType));
         }
+        if (result.verified) {
+            document.add(new Field(extractionResultField, EXTRACTION_COMPLETE, fieldType));
+        } else {
+            document.add(new Field(extractionResultField, EXTRACTION_PARTIAL, fieldType));
+        }
+        if (result.alwaysMatch) {
+            document.add(new Field(extractionResultField, EXTRACTION_ALWAYS_MATCH, fieldType));
+        }
```

Comment:
I think we can do this, but this does mean that in PercolateQuery.java line 87 we would need to use a `TermsQuery` instead of a `TermQuery` for the `verifiedQueriesQuery` so that we skip memory index verification for percolator queries containing only match_all queries.